fix: link hover underline styling

Xavi1 opened the pull request:
- Applied global CSS to underline all <a> tags on hover
- Removed default underlines from menu links
- Ensured CRUD menu links match updated styling

filiptrivan reviewed (changes requested):
1. Put the:
```
    &:hover {
      background-color: var(--p-surface-800);
      text-decoration: none;
    }
```
inside `.layout-menuitem`, also you forgot to add it on the playground demo app menu items.

2. Don't add `no-underline` class anywhere, I will solve it by merging `icon-link` and `side-menu-link`

3. Gread job noticing the `p-menu`, I would just put the fixer globally, because maybe I would use `p-menu` somewhere else tomorrow and I will forget about this. Just do it without `::ng-deep` and `.no-underline-menu`.

4.

Xavi1 commented:
Thanks for the feedback!

- Applied comprehensive global styles to remove text-decoration on hover for all PrimeNG menu components

- Updated header component to explicitly set text-decoration: none on hover for nav links

- Added specific overrides for context menus and related PrimeNG components

- Ensured consistent styling across all menu types: main nav, sidebar, and context menus

- Removed all .no-underline class usage and avoided ::ng-deep as suggested
